stl/inc/tuple

```diff
@@ -888,19 +1060,52 @@ _NODISCARD constexpr _Ty make_from_tuple(_Tuple&& _Tpl) { // construct _Ty from
 template <class... _Types, class _Alloc>
 struct uses_allocator<tuple<_Types...>, _Alloc> : true_type {}; // true_type if container allocator enabled
 
-#ifdef __cpp_lib_concepts
-template <class... _TTypes, class... _UTypes, template <class> class _TQual, template <class> class _UQual>
-    requires requires { typename tuple<common_reference_t<_TQual<_TTypes>, _UQual<_UTypes>>...>; }
-struct basic_common_reference<tuple<_TTypes...>, tuple<_UTypes...>, _TQual, _UQual> {
-    using type = tuple<common_reference_t<_TQual<_TTypes>, _UQual<_UTypes>>...>;
+#if _HAS_CXX23 && defined(__cpp_lib_concepts)

```

Comment:
No change requested here. But it would be nice to fix the condition in `<utility>`.

https://github.com/microsoft/STL/blob/1868694e98564bcf774dc542c14df185ec0942d0/stl/inc/utility#L526

Condition in this line should also be `_HAS_CXX23 && defined(__cpp_lib_concepts)`. Currently these partial specializations are provided in C++20 mode, which is probably incorrect.

Reply:
OK, I'm going to fix this later.